src/GafferUI/Handle.cpp

```diff
@@ -575,17 +575,17 @@ Handle::AngularDrag::AngularDrag( const Gadget *gadget, const Imath::V3f &origin
 		// Looking down the normal of the rotation plane, positive values rotate
 		// counterclockwise. Find the gadget-space vector pointing in the direction
 		// of positive rotation to use for the start direction of the linear drag.
-		V3f parallelAxis = dragAxis == 0 ? axis90 : axis0;
-		float parallelDot = dragAxis == 0 ? axis90Dot : axis0Dot;
+		V3f positiveDirection = dragAxis == 0 ? axis0 : axis90;
 
 		// If the parallel axis is pointing in the same direction as the view,
 		// reverse the drag direction.
-		float sign = -( ( 0 < parallelDot ) - ( parallelDot < 0 ) );
+		float parallelDot = dragAxis == 0 ? -axis90Dot : axis0Dot;
```

Comment:
I don't quite understand the underlying math of why this negation is necessary unfortunately. I arrived at that result by charting out the different axis configurations and seeing which needed to be flipped, along with the `sign` below. I can certainly revise if there's a more elegant solution I'm missing.

Reply:
@danieldresser-ie, could you take a look at this please? Since you reviewed originally I think you have a better chance of evaluating the new code...

Reply:
I tried to figure out why the negative sign is necessary here, and didn't quite figure it out. But this approach seems to yield the same result, and I think is simpler?

```    if( abs( dragData.worldNormal.dot( dragData.worldLine.dir ) ) < g_planarToLinearDragThreshold )
    {
        // Do a LinearDrag in raster space instead of world space

        // We're going to find a direction in raster space where all movement along this axis will
        // be treated as rotation. We want a direction in the 2d plane that is perpendicular to the
        // normal, which we can get via cross product between the normal and the view direction.
        V3f worldRotationDirection = dragData.worldLine.dir.cross( dragData.worldNormal );

        // Convert to raster space
        const ViewportGadget *viewport = gadget->ancestor<ViewportGadget>();
        V2f rasterRotationDirection =
            ( viewport->worldToRasterSpace( worldRotationDirection + dragData.worldOrigin )
            - viewport->worldToRasterSpace( dragData.worldOrigin ) ).normalized()
            * V2f( 1.f, -1.f ) // Y is positive down in raster space, flip the Y direction
        ;

        m_drag = LinearDrag(
            gadget,
            rasterRotationDirection,
            dragBeginEvent,
            m_processModifiers
        );
        m_dragBeginRotation = std::get<LinearDrag>( m_drag ).startPosition();
    }
```

Reply:
That is much simpler and I confirmed that the rotation directions are right for all the directions I can think of to test.  Thanks Daniel!

Fixed in https://github.com/GafferHQ/gaffer/pull/5281/commits/154226b61ef5cd56754c803d42794b07fcec1131